backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -62,4 +64,12 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 	public Map<UUID, Shop> getAllShops() {
 		return shopMap;
 	}
+
+	public List<Token> getTokens(UUID shopId) {
+		if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+			return shopIdToListOfTokensMap.get(shopId).stream()
+					.filter(token -> token.getStatus() == Status.ACTIVE).collect(Collectors.toList());
+		}
+		return null;
```

Comment:
In general, we should avoid sending null. We can send an empty list, and log an error saying "Tried to fetch tokens of a shop which is not in the ACTIVE state."

Reply:
Done!